### PR TITLE
Updated checkout version tag

### LINKED ISSUE #8334: Actions Deprecation Warning: NodeJS
### rAthena Hash

a9a8b0d

### Client Date

N/A

### Server Mode

Pre-Renewal, Renewal

### Result

In all Github Actions where we use actions/checkout at a lower versioned tag than v3 a deprecation notice is displayed.

### Relevant Log Output

```Shell
The following actions uses node12 which is deprecated and will be forced to run on node16: actions/checkout@v2. For more info: https://github.blog/changelog/2023-06-13-github-actions-all-actions-will-run-on-node16-instead-of-node12-by-default/
```

### Expected Result

No deprecation warning..

### How to Reproduce

Run any of our Actions.

### Official Information

N/A

### Modifications that may affect results

_No response_

### PR DESCRIPTION
**Pull request by Akkarinage**
* **Addressed Issue(s)**: #8334 

* **Server Mode**: Both

* **Description of Pull Request**: 
* Updates actions/checkout to version 4 which uses a newer version of NodeJS.
* This PR resolves deprecation and fixes #8334

### CODE REVIEW

File: .github/workflows/analysis_codeql.yml [hunk 1]
```diff
@@ -74,7 +74,7 @@ jobs:
       #- name: Autobuild
       #  uses: github/codeql-action/autobuild@v2
 
-      # ✏️ If the Autobuild fails above, remove it and uncomment the following
+      # ✝︝ If the Autobuild fails above, remove it and uncomment the following
```
Comment (aleos89) [inline .github/workflows/analysis_codeql.yml:77]: Looks like the icon Unicode broke.

**Review by aleos89 — APPROVED** [review]

**Comment by Akkarinage** [comment]
Missed an update on the Windows Action - on mobile and can't lookup the new version. Will update tomorrow unless someone wants to take a stab at it?

File: .github/workflows/analysis_codeql.yml (hunk 1 above)
Comment (Akkarinage) [reply .github/workflows/analysis_codeql.yml:77]: 🤦‍♂️

Comment (aleos89) [reply .github/workflows/analysis_codeql.yml:77]: It could just be GitHub tricking us!

**Comment by Akkarinage** [comment]
There we go, no more deprecation notices.

**Review by aleos89 — APPROVED** [review]

**Pull request merged**